docs/tutorial/using-grpc-plugins.mdx

```diff
@@ -0,0 +1,473 @@
+```
+
+You should now have a directory containing a single plugin, `starwars`. We’ll go over what each file does soon.
+
+```
+starwars
+├── Dockerfile
+├── generated
+│   ├── mapping.json
+│   ├── service.pb.go
+│   ├── service.proto
+│   ├── service.proto.lock.json
+│   └── service_grpc.pb.go
+├── go.mod
+├── go.sum
+├── README.md
+└── src
+    ├── main.go
+    ├── main_test.go
+    └── schema.graphql
+```
+
+The first thing we need to do is take a look at the GraphQL schema in `starwars/src/schema.graphql`. It doesn’t contain our schema yet, so we’ll start by defining our new service in GraphQL terms.
+
+When you open this file, it will have some placeholder schema inside. You can safely remove all of the content and replace it with the following:
+
+```
+type Person {
+  """
+  The name of this person
+  """
+  name: String!
+
+  """
+  The height of the person in centimeters
+  """
+  height: String!
+
+  """
+  The mass of the person in kilograms
+  """
+  mass: String!
+
+  """
+  The hair color of this person. Will be "unknown" if not known or "n/a" if the person does not have hair
+  """
+  hair_color: String!
+
+  """
+  The skin color of this person
+  """
+  skin_color: String!
+
+  """
+  The eye color of this person. Will be "unknown" if not known or "n/a" if the person does not have an eye
+  """
+  eye_color: String!
+
+  """
+  The birth year of the person, using BBY or ABY (Before/After Battle of Yavin)
+  """
+  birth_year: String!
+
+  """
+  The gender of this person. Either "Male", "Female" or "unknown", "n/a" if no gender
+  """
+  gender: String!
+}
+
+type Query {
+  """
+  get all the people
+  """
+  people: [Person!]!
+
+  """
+  get a specific person
+  """
+  person(id: ID!): Person!
+}
+```
+
+For this example, we won’t wrap the entire SWAPI, only the `people` resource and its endpoints.
+
+This schema has a single type, “Person”, and a query to get all the people or a specific person by ID.
+
+## Generate and Implement
+
+Now we can use `wgc` again to generate the Protobuf representation of our subgraph and boilerplate code to implement it in Golang.
+
+```bash
+wgc router plugin generate ./starwars
+```
+
+You’ll now see a few new files in the `generated` folder.
+
+```
+generated
+├── mapping.json
+├── service.pb.go
+├── service.proto
+├── service.proto.lock.json
+└── service_grpc.pb.go
+```
+
+In short, these files are generated helpers based on the schema we wrote. They help either translate GraphQL operations to gRPC or let you write type-safe resolvers in the plugin itself.
+
+
+<Note>
+  We recommend checking this folder into version control (e.g. Git).
+</Note>
+
+Now, let’s start implementing our resolvers. After generating, if you open `main.go`, you will see some errors about undefined types. These are remnants from the example schema’s resolver, and you can safely delete them to start from scratch.
+
+Here’s a starting point for implementing our new resolvers:
+
+```go
+type StarwarsService struct {
```

Comment:
The file needs to fully functional. Assume non go developers are trying to make this work.

Reply:
The tutorial guides through the full implementation step by step, this is just a starting point to follow along